src/bundle/Resources/public/js/scripts/admin.notifications.modal.js

```diff
@@ -1,89 +1,171 @@
+    };
+    const updateUserAvatar = (notificationsInfo) => {
+        const pendingNotificationsExist = notificationsInfo.pending;
+        const userAvatar = doc.querySelector('.ez-user-menu__avatar-wrapper');
+
+        userAvatar.dataset.count = notificationsInfo.pending;
+        userAvatar.classList.toggle('n-pending-notifications', pendingNotificationsExist);
+
+        doc.querySelector('.ez-user-menu__item--notifications').dataset.count = `(${notificationsInfo.pending})`;
+    };
+    const setPendingNotificationCount = (notificationsInfo) => {
+        updateUserAvatar(notificationsInfo);
+
+        const notificationsTable = modal.querySelector(SELECTOR_TABLE);
+        const notificationsTotal = notificationsInfo.total;
+        const notificationsTotalOld = parseInt(notificationsTable.dataset.notificationsTotal, 10);
+
+        if (notificationsTotal !== notificationsTotalOld) {
+            notificationsTable.dataset.notificationsTotal = notificationsTotal;
+
+            refreshNotificationsPage();
+        }
     };
-    const getNotificationsList = () => {
-        const requestOptions = {
+    const setNotificationPage = (pageHtml) => {
+        const modalResults = modal.querySelector(SELECTOR_MODAL_RESULTS);
+
+        modalResults.innerHTML = pageHtml;
+        toggleLoading(true);
+    };
+    const toggleSpinner = (force) => {
+        const spinner = modal.querySelector(SELECTOR_MODAL_SPINNER);
+
+        spinner.classList.toggle(CLASS_MODAL_SPINNER_INVISIBLE, force);
+    };
+    const toggleResults = (force) => {
+        const results = modal.querySelector(SELECTOR_MODAL_RESULTS);
+
+        results.classList.toggle(CLASS_MODAL_RESULTS_INVISIBLE, force);
+    };
+    const toggleLoading = (force) => {
+        toggleSpinner(force);
+        toggleResults(!force);
+    };
+    const fetchNotificationPage = (link) => {
+        if (!link) {
+            return;
+        }
+
+        const request = new Request(link, {
+            method: 'GET',
             headers: {
-                'X-CSRF-Token': token
+                Accept: 'text/html',
             },
+            credentials: 'same-origin',
             mode: 'cors',
-            credentials: 'same-origin'
-        };
+        });
 
-        fetch(modal.querySelector(SELECTOR_TABLE).dataset.notifications, requestOptions)
-            .then(response => response.text())
-            .then(notifications => doc.querySelector(SELECTOR_TABLE_BODY).innerHTML = notifications);
+        currentPageLink = link;
+        toggleLoading(false);
+        fetch(request)
+            .then(handleRequestResponseText)
+            .then(setNotificationPage)
+            .catch(handleResponseError);
     };
-    const setPendingNotificationCount = (notificationsInfo) => {
-        const methodName = notificationsInfo.pending ? 'add' : 'remove';
-        const userAvatar = doc.querySelector('.ez-user-menu__avatar-wrapper');
+    const refreshNotificationsPage = () => {
```

Comment:
I don't see any reason to have this wrapper function. This is called once and you could call `fetchNotificationPage(currentPageLink);` directly.